### DISCUSSION TITLE
Gatsby MDX reports invalid AST

### DISCUSSION THREAD

**Discussion by davwheat** [gatsbyjs/gatsby#37518] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

When using the MDX plugin and programmatically generating pages, I get an error about invalid AST which renders MDX-generated pages unusable.

### Reproduction Link

https://github.com/davwheat/gatsby-issue-reproduction-3

**MDX content:**

```mdx
---
title: "Test?"
description: Test!
path: test
created_at: 2022-04-19T16:00Z
---

EE's innovation in RAN has found it at the bleeding edge of new 5G deployments in the UK, being the first operator to deploy 5G NR carrier
aggregation in a relatively widespread manner across many parts of the UK capital, London.

```

### Steps to Reproduce

1. Clone repo
2. Run `yarn develop`

### Expected Result

The pages to be built in SSG successfully.

### Actual Result

SSG fails due to "invalid AST".

### Environment

```shell
$ /home/david/gatsby-mdx-test/node_modules/.bin/gatsby info --clipboard

  System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (6) x64 12th Gen Intel(R) Core(TM) i7-12700H
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.13.0 - /tmp/yarn--1674431542087-0.36198500852724314/node
    Yarn: 1.22.19 - /tmp/yarn--1674431542087-0.36198500852724314/yarn
    npm: 8.19.3 - /usr/bin/npm
  npmPackages:
    gatsby: ^5.4.2 => 5.4.2
    gatsby-plugin-mdx: ^5.4.0 => 5.4.0
    gatsby-source-filesystem: ^5.4.0 => 5.4.0


Done in 2.99s.
```

### Config Flags

Happens with DEV SSR on (in project) and off (in reproduction).

**Comment by davwheat** [comment]
On further inspection this suggestions that Typescript is not supported with MDX? The line and position for the error shows it to be L11 C35, which is `export interface IMdxPageContext {`.

I don't think this is mentioned in the Gatsby docs at all.

**Comment by PaulieScanlon** [comment]
Hey @davwheat i had a look at your test.mdx file. You have a couple of unescaped characters that MDX2 will throw on. The MDX error is usually quite misleading and points to a line in the file that isn't the actual problem. 

I wrote this post a while ago. Maybe try a test by removing any "special" characters. I've linked to the breaking changes in the MDX docs in the post.

https://paulie.dev/posts/2022/08/mdx-2-breaking-changes-and-gatsby-plugin-mdx-v4/

**Comment by davwheat** [comment]
Thanks for your suggestion!

I hadn't realised this, and have changed my `test.mdx` file accordingly to contain only a minimal example, which still results in the same error for me, unfortunately.
